tests/ClientTest.php

```diff
@@ -0,0 +1,292 @@
+    {
+        $mock = new MockHandler([
+            new Response(
+                500,
+                ['Content-Type' => 'application/json'],
+                '{"message" => "Out of order"}'
+            ),
+            new Response(
+                501,
+                ['Content-Type' => 'application/json'],
+                'Out of order'
+            ),
+            new Response(
+                200,
+                ['Content-Type' => 'application/json'],
+                '{
+                    "id": "683194249",
+                    "runId": "683194249",
+                    "status": "created",
+                    "desiredStatus": "processing",
+                    "mode": "run",
+                    "component": "keboola.ex-db-snowflake",
+                    "config": "123",
+                    "configRow": null,
+                    "tag": null,
+                    "createdTime": "2021-03-04T21:59:49+00:00"
+                }'
+            ),
+        ]);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $client = $this->getClient(['handler' => $stack]);
+        $client->postEvent($this->getPostEventData());
+        self::assertCount(3, $requestHistory);
+        /** @var Request $request */
+        $request = $requestHistory[0]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+        $request = $requestHistory[1]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+        $request = $requestHistory[2]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+    }
+
+    public function testRetryFailure(): void
+    {
+        $responses = [];
+        for ($i = 0; $i < 30; $i++) {
+            $responses[] = new Response(
+                500,
+                ['Content-Type' => 'application/json'],
+                '{"message" => "Out of order"}'
+            );
+        }
+        $mock = new MockHandler($responses);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $client = $this->getClient(['handler' => $stack, 'backoffMaxTries' => 1]);
+        try {
+            $client->postEvent($this->getPostEventData());
+            self::fail('Must throw exception');
+        } catch (ClientException $e) {
+            self::assertStringContainsString('500 Internal Server Error', $e->getMessage());
+        }
+        self::assertCount(2, $requestHistory);
+    }
+
+    public function testRetryFailureReducedBackoff(): void
```

Comment:
V cem se tenhle test lisi od prechoziho? Jediny rozdil vidim v poctu retries. Hraje to nejakou roli?

Reply:
Ne už ne, zamýšlel jsem to jinak, mažu